oauth2_provider/tests/test_authorization_code.py

```diff
@@ -908,6 +909,139 @@ def test_code_exchange_succeed_when_redirect_uri_match(self):
         self.assertEqual(content['scope'], "read write")
         self.assertEqual(content['expires_in'], oauth2_settings.ACCESS_TOKEN_EXPIRE_SECONDS)
 
+    def test_pkce(self):
+        self.client.login(username="test_user", password="123456")
+        hashed_data = hashlib.sha256(b'secret').hexdigest()
+
+        # retrieve a valid authorization code
+        authcode_data = {
+            'client_id': self.application.client_id,
+            'state': 'random_state_string',
+            'scope': 'read write',
+            'redirect_uri': 'http://example.it?foo=bar',
+            'response_type': 'code',
+            'allow': True,
+            'pkce': hashed_data
```

Comment:
I think we should call this `code_challenge`, that's what the spec says. The less deviation the better, how many unique oauth implementations have already been unleashed on the world?